service/src/test/java/uk/gov/hmcts/reform/fpl/service/DraftCMOServiceTest.java

```diff
@@ -92,6 +95,50 @@ void shouldReturnCaseManagementOrderWhenProvidedCaseDetails() {
             formatLocalDateToMediumStyle(5));
     }
 
+    @Test
+    void shouldFormatRespondentsIntoKeyWhenRespondentsArePresent() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(ImmutableMap.of("respondents1", createRespondents()))
+            .build();
+
+        String respondentsKey = draftCMOService.createRespondentAssigneeDropdownKey(caseDetails);
+
+        assertThat(respondentsKey).contains(
+            "Respondent 1 - Timothy Jones",
+            "Respondent 2 - Sarah Simpson");
+    }
+
+    @Test
+    void shouldReturnEmptyStringWhenRespondentsAreNotPresent() {
+        CaseDetails caseDetails = CaseDetails.builder().data(ImmutableMap.of()).build();
+
+        String respondentsKey = draftCMOService.createRespondentAssigneeDropdownKey(caseDetails);
+
+        assertThat(respondentsKey).isEqualTo("");
+    }
+
+    @Test
+    void shouldFormatOthersIntoKeyWhenOthersArePresent() {
```

Comment:
What about test case where name is null or empty?

Reply:
Added additional test to assert if an other contains no full name then the empty placeholder of "BLANK - please complete" will appear.